macOS(error: use of undeclared identifier 'MSG_MORE')

On macOS, build fails in `send.c` because `MSG_MORE` is not defined for Darwin. Running the build errs at the `send()` call, so the library can't be built without a fallback.

Error log:
```log
$ cmake -S . -B build && cmake --build build
...
/Users/darwin/dragonnet/dragonnet/send.c:10:67: error: use of undeclared identifier 'MSG_MORE'
   10 |         ssize_t len = send(p->sock, buf, n, MSG_NOSIGNAL | (submit ? 0 : MSG_MORE));
      |                                                                          ^
1 error generated.
gmake[2]: *** [CMakeFiles/dragonnet.dir/build.make:177: CMakeFiles/dragonnet.dir/dragonnet/send.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/dragonnet.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
```

--- a/dragonnet/sock.h
+++ b/dragonnet/sock.h
@@ -15,4 +15,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <netdb.h>
+#ifndef MSG_MORE
+#define MSG_MORE 0
+#endif
 #endif

